instancio-junit/src/main/java/org/instancio/junit/internal/InstancioArgumentsProvider.java

```diff
@@ -57,39 +52,21 @@ public Stream<? extends Arguments> provideArguments(final ExtensionContext conte
         final Random random = threadLocalRandom.get();
         final Settings settings = threadLocalSettings.get();
         final Type[] paramTypes = context.getRequiredTestMethod().getGenericParameterTypes();
-        final Object[] args = createObjectsGroupingByType(paramTypes, random, settings);
+        final Object[] args = createObjects(paramTypes, random, settings);
         return args.length == 0 ? Stream.of() : Stream.of(Arguments.of(args));
     }
 
-    /*
-     * Since the seed value is the same for all parameters, parameters of the same type
-     * should be generated using a single context. This is to prevent the same value being
-     * generated, e.g. @InstancioSource({String.class, String.class}) => "foo", "foo"
-     */
-    static Object[] createObjectsGroupingByType(final Type[] types, final Random random, final Settings settings) {
-        final Map<Type, Long> counts = Arrays.stream(types)
-                .collect(Collectors.groupingBy(Function.identity(), Collectors.counting()));
-
-        final Map<Type, Queue<Object>> resultsByType = new LinkedHashMap<>();
-
-        counts.forEach((type, count) -> {
-            final InstancioApi<?> api = Instancio.of(() -> type);
-            if (settings != null) {
-                api.withSettings(settings);
-            }
-            final Queue<Object> results = api
-                    .withSeed(random.getSeed())
-                    .stream()
-                    .limit(count)
-                    .collect(Collectors.toCollection(ArrayDeque::new));
-
-            resultsByType.put(type, results);
-        });
+    static Object[] createObjects(final Type[] types, final Random random, final Settings settings) {
+        return Arrays.stream(types).map(type -> createObject(type, random, settings)).collect(Collectors.toList()).toArray();
+    }
 
-        final Object[] results = new Object[types.length];
-        for (int i = 0; i < results.length; i++) {
-            results[i] = resultsByType.get(types[i]).poll();
+    static Object createObject(final Type type, final Random random, final Settings settings) {
+        final InstancioApi<?> api = Instancio.of(() -> type);
+        if (settings != null) {
+            api.withSettings(settings);
         }
-        return results;
+        return api
+                .withSeed(random.longRange(Long.MIN_VALUE, Long.MAX_VALUE))
```

Comment:
Only one minor comment: let's use `1` as the min seed value. By default Instancio generates positive seeds to avoid copy/paste erros (e.g. user copy-pastes reported seed into the `@Seed` annotation, but accidentally omits the minus).

Reply:
That's a really good idea!